analyzer/tests/functional/ctu_failure/test_ctu_failure.py

```diff
@@ -146,6 +167,60 @@ def check_source_in_archive(source_in_archive):
             check_source_in_archive("main.c")
             check_source_in_archive("lib.c")
 
+    def test_ctu_on_demand_failure_zip(self):
+        """ Test the failure zip contains the source of imported TU when using
+        on-demand mode.
+        """
+        self.__set_up_test_dir('ctu_failure')
```

Comment:
We use the same directory name as in the `test_ctu_on_demand_logs_ast_import` function. Isn't it a problem?

Reply:
This might be a problem. To be safe I will use another prefix/postfix for this test.